src/displayapp/screens/TwosGrid.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <displayapp/TouchEvents.h>
+
+namespace Pinetime {
+  namespace Applications {
+    namespace Screens {
+      struct TwosTile {
+        bool merged = false;
+        unsigned int value = 0;
+      };
+
+      class TwosGrid {
+      public:
+        TwosGrid();
+        ~TwosGrid() = default;
+
+        static constexpr int nCols = 4;
+        static constexpr int nRows = 4;
+        static constexpr int nCells = nCols * nRows;
+
+        unsigned int getValue(unsigned int row, unsigned int col);
+        bool onEvent(
+          Pinetime::Applications::TouchEvents,
+          unsigned int* score);
+        bool placeNewTile();
+
+      private:
+        TwosTile grid[nRows][nCols];
+        bool tryMerge(int newRow, int newCol, int oldRow, int oldCol, unsigned int* score);
+        bool tryMove(int newRow, int newCol, int oldRow, int oldCol);
```

Comment:
These functions should use PascalCase (i.e. the first letters should be capitalised).